### PR TITLE
correct index offset in stacked bar charts with Brush component

### PR DESCRIPTION
**Pull request by 37108**
## Description

When using a Brush component with stacked bar charts, there was an index mismatch issue that caused incorrect data to be displayed. When the brush was moved to filter data (e.g., showing only the last 3 pages), the chart would either display wrong data or disappear completely.

The root cause was that `displayedData` was sliced to match the brush selection (e.g., indices 0, 1, 2 for the last 3 pages), but `stackedData` contained all 6 original data points. When rendering, the code accessed `stackedData[0]` thinking it was Page E data, but it actually contained Page A data.

Added `dataStartIndex` parameter to the bar rendering pipeline to properly

offset the index when accessing the full `stackedData` array. Now the code

uses `stackedData[index + dataStartIndex]` to access the correct element.

- `src/state/selectors/barSelectors.ts`: Pass `dataStartIndex` to `computeBarRectangles` function
- `src/cartesian/Bar.tsx`: Add `dataStartIndex` parameter and use it to offset stackedData access (line 744: `stackedData[index + dataStartIndex]`)
- `storybook/stories/Examples/BarChart/BarChart.stories.tsx`: Add `StackedWithBrush` story to demonstrate the fix
- `test/cartesian/Brush.stacked.spec.tsx`: Add comprehensive tests for brush behavior with stacked charts, including edge cases when narrowing to last 3 pages

All 6 tests in Brush.stacked.spec.tsx pass, including new tests that verify:
- Bars render correctly when brush is narrowed to last 3 pages
- Stack groups maintain correct data structure throughout filtering

## Related Issue
- https://github.com/recharts/recharts/issues/6331

## Motivation and Context
To fix a bug.

## How Has This Been Tested?

- Added comprehensive test suite in `test/cartesian/Brush.stacked.spec.tsx`:
  - Tests for Brush interaction with stacked Bar charts
  - Verifies correct bar rendering after moving the Brush traveller
  - Validates stack groups and stacked data structure with proper data range
  - Tests both initial view and narrowed down view (last 3 pages)
- Added Storybook story `StackedWithBrush` in `storybook/stories/Examples/BarChart/BarChart.stories.tsx` for manual testing and visual verification
- Verified the fix resolves the issue reported in #6006

## Screenshots (if appropriate):

https://github.com/user-attachments/assets/2364c811-883b-4c70-8065-7dd17b7e8c96

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] I have added a storybook story or VR test, or extended an existing story or VR test to show my changes

### CODE REVIEW

File: src/cartesian/Bar.tsx [hunk 1]
```diff
@@ -738,8 +740,8 @@ export function computeBarRectangles({
       let value, x: number | null, y, width, height, background: Rectangle;
 
       if (stackedData) {
-        // we don't need to use dataStartIndex here, because stackedData is already sliced from the selector
```
Comment (PavelVanecek) [inline src/cartesian/Bar.tsx:741]: What is the situation here? The old comment seems to contradict the PR.

Comment (ckifer) [reply src/cartesian/Bar.tsx:741]: right - did it become un-sliced in the selector? If so, should we fix it there instead?

Comment (ckifer) [reply src/cartesian/Bar.tsx:741]: from what I can tell stackedData was never sliced in the selector

Comment (37108) [reply src/cartesian/Bar.tsx:741]: The old comment was wrong. stackedData was never sliced in the selector (only displayedData was sliced at line 527). That's exactly why this fix is needed, we must use index + dataStartIndex to access the correct element in the full (non-sliced) stackedData array.

**Comment by PavelVanecek** [comment]
We merged this thing in June 2025: https://github.com/recharts/recharts/pull/6009 did it break since?

**Review by PavelVanecek — APPROVED** [review]

**Comment by ckifer** [comment]
That PR never re-slices the data even though it says that it does. Not sure how things were working

**Comment by mariuskuehnast** [comment]
Thank you for the fix ❤️
